interfaces/cython/cantera/examples/reactors/reactor1.py

```diff
@@ -12,16 +12,26 @@
 r = ct.IdealGasConstPressureReactor(gas)
 
 sim = ct.ReactorNet([r])
+sim.initialize()
+sim.verbose = True
+
+# limit advance when temperature difference is exceeded
+delta_T_max = 20.
+r.set_advance_limit('temperature', delta_T_max)
+
 time = 0.0
 states = ct.SolutionArray(gas, extra=['t'])
 
-print('%10s %10s %10s %14s' % ('t [s]','T [K]','P [Pa]','u [J/kg]'))
+print('{:10s} {:10s} {:10s} {:14s}'.format('t [s]','T [K]','P [Pa]','u [J/kg]'))
 for n in range(100):
     time += 1.e-5
-    sim.advance(time)
+    while sim.advance(time) < time:
+        states.append(r.thermo.state, t=sim.time*1e3)
+        print('{:10.3e} {:10.3f} {:10.3f} {:14.6f}'.format(sim.time, r.T,
+                                                           r.thermo.P, r.thermo.u))
```

Comment:
I realize it provides slightly different output, but I think the simpler integration loop that users will more often want to emulate would be one that limits the maximum Δt and maximum ΔT, but without necessarily needing outputs at a pre-defined set of times, e.g.
```python
while sim.time < t_end:
    sim.advance(sim.time + dt_max)
    states.append(...)
    print(...)
```

Reply:
changed.